sessiontxn/isolation/base.go

```diff
@@ -337,6 +337,7 @@ func (p *baseTxnContextProvider) ActivateTxn() (kv.Transaction, error) {
 	if sessVars.LoadBasedReplicaReadThreshold > 0 {
 		txn.SetOption(kv.LoadBasedReplicaReadThreshold, sessVars.LoadBasedReplicaReadThreshold)
 	}
+	txn.SetOption(kv.ResourceGroupName, sessVars.ResourceGroupName)
```

Comment:
I noticed `StalenessTxnContextProvider`'s `ActivateTxn`, do we need to SetOption also?

Reply:
fixed